src/drivers/intel/cavs/dmic.c

```diff
@@ -1170,6 +1172,7 @@ static int dmic_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 	switch (dmic_prm[di]->fifo_bits) {
 	case 0:
 	case 16:
+	case 24:
```

Comment:
Just curious, and can be fixed late if needed, but why is 0 treated in the switch alongside supported and valid formats ?

Reply:
I recall the kernel passed/driver stored configuration contains zero in fifo_bits if we have not set up a FIFO even if it exists in HW. This avoids the driver to not stop global all FIFOs related configuration in such case.

Reply:
ok, thanks for the explanation - it's worth a comment in a subsequent patch.